spec/requests/api/v1/targets/create_spec.rb

```diff
@@ -78,4 +78,28 @@
       expect { subject }.not_to change { Target.count }
     end
   end
+
+  context 'when the user is not authenticated' do
+    subject { post api_v1_targets_url, params: target_params, as: :json }
```

Comment:
You can do 
`subject { post api_v1_targets_url, params: target_params, headers: headers, as: :json }`

And then in each context 
`let(:headers) { auth_headers }  `  or   `let(:headers) { nil }  ` depending if you want the user to be logged in or not

Reply:
+1 on this one. It will become very useful and keep your code DRYer.

Reply:
Thanks!